test/e2e/e2e_test.go

```diff
@@ -891,47 +1042,226 @@ func deleteRC(ctx context.Context, t *testing.T, clientSet clientset.Interface,
 	}
 }
 
-func evictPods(ctx context.Context, t *testing.T, clientSet clientset.Interface, nodeInformer coreinformers.NodeInformer, nodeList []*v1.Node, rc *v1.ReplicationController) {
-	var leastLoadedNode *v1.Node
-	podsBefore := math.MaxInt16
-	evictionPolicyGroupVersion, err := eutils.SupportEviction(clientSet)
-	if err != nil || len(evictionPolicyGroupVersion) == 0 {
-		klog.Fatalf("%v", err)
+var balancePodLabel = map[string]string{"podname": "priority-balanced-memory"}
+
+// track min memory limit based on crio minimum. pods cannot set a limit lower than this
+// see: https://github.com/cri-o/cri-o/blob/29805b13e9a43d9d22628553db337ce1c1bec0a8/internal/config/cgmgr/cgmgr.go#L23
+// see: https://bugzilla.redhat.com/show_bug.cgi?id=1595256
+var crioMinMemLimit = 12 * 1024 * 1024
+
+var podRequestedResource = &v1.ResourceRequirements{
+	Limits: v1.ResourceList{
+		v1.ResourceMemory: resource.MustParse("100Mi"),
+		v1.ResourceCPU:    resource.MustParse("100m"),
+	},
+	Requests: v1.ResourceList{
+		v1.ResourceMemory: resource.MustParse("100Mi"),
+		v1.ResourceCPU:    resource.MustParse("100m"),
+	},
+}
+
+// createBalancedPodForNodes creates a pod per node that asks for enough resources to make all nodes have the same mem/cpu usage ratio.
+func createBalancedPodForNodes(
```

Comment:
Maybe this could be moved to `component-helpers`, since it currently lives in https://github.com/kubernetes/kubernetes/blob/84483a5/test/e2e/scheduling/priorities.go#L478 and is used in a couple places. Could be a good external scheduling test util function to have

Reply:
It turned out this `createBalancedPodForNodes` is not the exact copy of the one in https://github.com/kubernetes/kubernetes/blob/84483a5/test/e2e/scheduling/priorities.go#L478. Priorities take [non-zero requests](https://github.com/kubernetes/kubernetes/blob/84483a5/test/e2e/scheduling/priorities.go#L626) whereas descheduler takes even zero requests. Also, `schedutil.GetNonzeroRequests` is still in [k/k repo](https://github.com/kubernetes/kubernetes/blob/84483a5aacba8e535c9e185e9b0c01a5248b5086/pkg/scheduler/util/non_zero.go#L42) including [DefaultMilliCPURequest and DefaultMemoryRequest](https://github.com/kubernetes/kubernetes/blob/84483a5aacba8e535c9e185e9b0c01a5248b5086/pkg/scheduler/util/non_zero.go#L35-L37) consts which can not be separated from the scheduling framework :-|.

Reply:
Ah, it's more complicated than I thought then. That's fine, this can be a TODO at a later time as we improve the exportability of some of these functions. Maybe we can add a comment about that function in k/k just as a reference of what this is based off of

Reply:
```
// TODO(jchaloup): The function is updated version of what under https://github.com/kubernetes/kubernetes/blob/84483a5/test/e2e/scheduling/priorities.go#L478.
//                 Import it once the function is moved under k8s.io/components-helper repo and modified to work for both priority and predicates cases.
```
added to the code

Reply:
Thanks!